src/components/common/input/index.js

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect } from "react";
+import "./index.scss";
+
+const Input = ({
+  type = "text",
+  className = "",
+  handleInputChange,
+  name,
+  value,
+  label,
+  errorLabel,
+  invalidErrorLabel,
+  regex,
+  checkValidation,
+  ...rest
+}) => {
+  const [error, setError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState("");
+  const [valueChanged, setValueChanged] = useState(false);
+  const regexValue = new RegExp(regex);
+
+  useEffect(() => {
+    validate();
+  }, [checkValidation]);
+
+  /**
+   * getError - handles errors
+   * @returns {object}
+   */
+  const getError = (e) => {
+    if (value.length === 0 && valueChanged && errorLabel) {
+      return {
+        error: true,
+        errorMsg: errorLabel,
+      };
+    } else if (value.length === 0 && checkValidation && errorLabel) {
+      return {
+        error: true,
+        errorMsg: errorLabel,
+      };
+    } else if (
+      value != "" &&
+      errorLabel &&
+      invalidErrorLabel &&
+      !regexValue.test(value)
+    ) {
+      return {
+        error: true,
+        errorMsg: invalidErrorLabel,
+      };
+    } else {
+      return {
+        error: false,

```

Comment:
use switch case and move the conditions in function which will return the case value